pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -52,11 +52,14 @@ const (
 )
 
 // AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+// +kubebuilder:validation:XValidation:rule="has(self.atlasRef) && !has(self.projectRef) ? true : !has(self.atlasRef) && has(self.projectRef)",message="cannot define both atlasRef and projectRef at same time"
```

Comment:
let's add CEL unit tests for this.

Reply:
xref https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1769#discussion_r1721226434

Reply:
For the CEL tests, I'd rather do that in a separate PR. This PR is big enough. In fact, we could probably merge the CEL test infra already, as it is not public facing, and just reuse here.